RecoVertex/KinematicFit/interface/KinematicConstrainedVertexUpdatorT.h

```diff
@@ -138,6 +139,11 @@ RefCountedKinematicVertex KinematicConstrainedVertexUpdatorT<nTrk, nConstraint>:
   //full math case now!
   val += g * delta_alpha;
   lambda = v_g_sym * val;
+  if (!edm::isFinite(lambda[0])) {
```

Comment:
in the problematic case here, was `v_g_sym` or `val` nan?
If `val` the check can be done earlier

Reply:
also, why is only `lambda[0]` checked; IIUC it's a matrix here (although it takes a mental effort to figure out the dimensions)

Reply:
also, what is the call stack trace? considering `lPoint: (-2.09345e+12,7.15679e+11,3.46131e+11)
fieldValue: (0,0,0) `, perhaps there is a missing check for a valid state earlier/upstream? Recall that the previous fix was to check for bz=0 in a different place in #39183

Reply:
> in the problematic case here, was `v_g_sym` or `val` nan?
> If `val` the check can be done earlier

If I interpret the printout from @smuzaffar in the original issue, when lamda is nan, val can either also be nan, have a ridiculous value, or have a sane value.